src/Symfony/Component/Console/DependencyInjection/AddConsoleCommandPass.php

```diff
@@ -86,6 +86,13 @@ public function process(ContainerBuilder $container)
             if ($aliases) {
                 $definition->addMethodCall('setAliases', [$aliases]);
             }
+
+            $storeDefinition = 'lock.default.store';
+            if ($container->hasDefinition($storeDefinition)) {
+                if (\method_exists($definition->getClass(), 'setStore')) {
```

Comment:
I'm a bit unsure about injecting the store based on method existence. 
This is about autoconfiguration which usually works on types, and traits are not types. Should we add a marker `LockableInterface` and use it as a type for `ContainerBuilder::registerForAutoconfiguration()`?

Reply:
That would mean that for every command I want to use configured stores I have to implement the `LockableInterface` and use the `LockableTrait`, correct? This in my opinion can cause problems if one forgets to implement the Interface or the trait.

Reply:
Right ... the user can also set `private $store;` already.Then, im not sure what we're solving here :thinking: 

We should favor pure DI / service usage IMHO.